src/DataProtection/custom/Cmdlets/Platform/Vault/Set-AzDataProtectionMSIPermission.ps1

```diff
@@ -0,0 +1,205 @@
+﻿function Set-AzDataProtectionMSIPermission {
+    [OutputType('Microsoft.Azure.PowerShell.Cmdlets.DataProtection.Models.Api20220401.IBackupInstanceResource')]
+    [CmdletBinding(PositionalBinding=$false)]
+    [Microsoft.Azure.PowerShell.Cmdlets.DataProtection.Description('Initializes Backup instance Request object for configuring backup')]
+
+    param(
+        [Parameter(Mandatory, HelpMessage='Backup instance request object which will be used to configure backup')]
+        [Microsoft.Azure.PowerShell.Cmdlets.DataProtection.Models.Api20220401.IBackupInstanceResource]
+        ${BackupInstance},
+        
+        [Parameter(Mandatory=$false, HelpMessage='ID of the keyvault')]
+        [ValidatePattern("/subscriptions/([A-z0-9\-]+)/resourceGroups/(?<rg>.+)/(?<id>.+)")]
+        [System.String]
+        ${KeyvaultId},
+
+        [Parameter(Mandatory, HelpMessage='Resource group of the backup vault')]
+        [System.String]
+        ${VaultResourceGroup},
+        
+        [Parameter(Mandatory, HelpMessage='Name of the backup vault')]
+        [System.String]
+        ${VaultName},
+
+        [Parameter(Mandatory, HelpMessage='Scope at which the permissions need to be granted')]
+        [System.String]
+        [ValidateSet("Resource","ResourceGroup","Subscription")]
+        ${PermissionsScope}
+        
+    )
+
+    process {
+          CheckResourcesModuleDependency
+          CheckPostgreSqlModuleDependency         
+          
+          $DatasourceId = $BackupInstance.Property.DataSourceInfo.ResourceId
+          $DatasourceType = $BackupInstance.Property.DataSourceInfo.ResourceType
+
+          if($DatasourceType -eq "Microsoft.Storage/storageAccounts"){$DatasourceType = "AzureBlob"}
+          elseif($DatasourceType -eq "Microsoft.Compute/disks"){$DatasourceType = "AzureDisk"}
+          elseif($DatasourceType -eq "Microsoft.DBforPostgreSQL/servers/databases"){$DatasourceType = "AzureDatabaseForPostgreSQL"}
+                 
+          $manifest = LoadManifest -DatasourceType $DatasourceType.ToString()
+
+          $ResourceArray = $DataSourceId.Split("/")
+          $ResourceRG = "/subscriptions/" + $ResourceArray[2] + "/resourceGroups/" + $ResourceArray[4]
+          $SubscriptionName = "/subscriptions/" + $ResourceArray[2]
```

Comment:
instead of doing this can we have a helper functions like 

GetSubscriptionIdFromArmId and GetResourceGroupIdFromArmId. As all the resources have a subscription and a resource group, these functions will make code for reusable and readable.